From e3b010273d894c83dca10fa98f8b4739b220413e Mon Sep 17 00:00:00 2001
From: "Vladimir D. Seleznev" <vseleznv@altlinux.org>
Date: Tue, 3 Sep 2019 21:20:52 +0300
Subject: [PATCH 5/9] ALT: build dependencies

---
 sqlite/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sqlite/Makefile.in b/sqlite/Makefile.in
index b2d066ea7..5e214e9fa 100644
--- a/sqlite/Makefile.in
+++ b/sqlite/Makefile.in
@@ -1152,7 +1152,7 @@ userauth.lo:	$(TOP)/ext/userauth/userauth.c $(HDR) $(EXTHDR)
 sqlite3session.lo:	$(TOP)/ext/session/sqlite3session.c $(HDR) $(EXTHDR)
 	$(LTCOMPILE) -DSQLITE_CORE -c $(TOP)/ext/session/sqlite3session.c
 
-json1.lo:	$(TOP)/ext/misc/json1.c
+json1.lo:	$(TOP)/ext/misc/json1.c $(HDR) $(EXTHDR)
 	$(LTCOMPILE) -DSQLITE_CORE -c $(TOP)/ext/misc/json1.c
 
 stmt.lo:	$(TOP)/ext/misc/stmt.c
-- 
2.25.4

