diff --git a/Makefile.in b/Makefile.in
index 5576de8..dc39016 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -749,7 +783,7 @@ tclsqlite3$(TEXE):	tclsqlite-shell.lo libsqlite3.la
 		 libsqlite3.la $(LIBTCL)
 
 testfixture$(TEXE):	$(TESTFIXTURE_SRC)
-	$(LTLINK) -DTCLSH=1 -DSQLITE_TEST=1 -DSQLITE_NO_SYNC=1\
+	$(LTLINK) -static -DTCLSH=1 -DSQLITE_TEST=1 -DSQLITE_NO_SYNC=1\
 		-DSQLITE_CRASH_TEST=1 \
                 -DSQLITE_SERVER=1 -DSQLITE_PRIVATE="" -DSQLITE_CORE $(TEMP_STORE) \
 		-o $@ $(TESTFIXTURE_SRC) $(LIBTCL) $(TLIBS)
