--- sqlite-src-3070400/Makefile.in.fts3	2010-12-07 19:24:21.000000000 +0300
+++ sqlite-src-3070400/Makefile.in	2011-01-25 10:18:23.237343665 +0300
@@ -26,7 +26,7 @@ BCC = @BUILD_CC@ @BUILD_CFLAGS@
 # will run on the target platform.  (BCC and TCC are usually the
 # same unless your are cross-compiling.)
 #
-TCC = @CC@ @CPPFLAGS@ @CFLAGS@ -I. -I${TOP}/src -I${TOP}/ext/rtree
+TCC = @CC@ @CPPFLAGS@ @CFLAGS@ -I. -I${TOP}/src -I${TOP}/ext/rtree -I${TOP}/ext/fts3
 
 # Define this for the autoconf-based build, so that the code knows it can
 # include the generated config.h
@@ -320,6 +320,17 @@ SRC += \
   $(TOP)/ext/fts3/fts3_tokenizer.c \
   $(TOP)/ext/fts3/fts3_tokenizer1.c \
   $(TOP)/ext/fts3/fts3_write.c
+OBJS0 += \
+  fts3.lo \
+  fts3_aux.lo \
+  fts3_expr.lo \
+  fts3_hash.lo \
+  fts3_icu.lo \
+  fts3_porter.lo \
+  fts3_snippet.lo \
+  fts3_tokenizer.lo \
+  fts3_tokenizer1.lo \
+  fts3_write.lo
 SRC += \
   $(TOP)/ext/icu/sqliteicu.h \
   $(TOP)/ext/icu/icu.c
