--- sqlite/Makefile.in.alt-fts3	2009-12-01 19:08:33 +0300
+++ sqlite/Makefile.in	2009-12-08 16:02:51 +0300
@@ -26,7 +26,7 @@ BCC = @BUILD_CC@ @BUILD_CFLAGS@
 # will run on the target platform.  (BCC and TCC are usually the
 # same unless your are cross-compiling.)
 #
-TCC = @CC@ @CPPFLAGS@ @CFLAGS@ -I. -I${TOP}/src
+TCC = @CC@ @CPPFLAGS@ @CFLAGS@ -I. -I${TOP}/src -I${TOP}/ext/fts3
 
 # Define this for the autoconf-based build, so that the code knows it can
 # include the generated config.h
@@ -323,6 +323,16 @@ SRC += \
   $(TOP)/ext/fts3/fts3_tokenizer.c \
   $(TOP)/ext/fts3/fts3_tokenizer1.c \
   $(TOP)/ext/fts3/fts3_write.c
+OBJS0 += \
+  fts3.lo \
+  fts3_expr.lo \
+  fts3_hash.lo \
+  fts3_icu.lo \
+  fts3_porter.lo \
+  fts3_snippet.lo \
+  fts3_tokenizer.lo \
+  fts3_tokenizer1.lo \
+  fts3_write.lo
 SRC += \
   $(TOP)/ext/icu/sqliteicu.h \
   $(TOP)/ext/icu/icu.c
