--- sqlite-3.7.4/Makefile.in-	2011-02-15 20:04:12.665295871 +0300
+++ sqlite-3.7.4/Makefile.in	2011-02-15 21:39:09.123249959 +0300
@@ -492,6 +492,7 @@
 
 libsqlite3.la:	$(LIBOBJ)
 	$(LTLINK) -o $@ $(LIBOBJ) $(TLIBS) \
+		-Wl,--version-script=libsqlite3.map -Wl,libsqlite3.lds -Wl,-z,defs \
 		${ALLOWRELEASE} -rpath "$(libdir)" -version-info "8:6:8"
 
 libtclsqlite3.la:	tclsqlite.lo libsqlite3.la
--- sqlite-3.7.4/libsqlite3.lds-	2011-02-15 21:35:18.763214945 +0300
+++ sqlite-3.7.4/libsqlite3.lds	2011-02-15 21:35:28.646359743 +0300
@@ -0,0 +1,191 @@
+"sqlite3AuthReadCol@SQLITE3_3.6.19" = sqlite3AuthReadCol;
+"sqlite3BtreeCheckpoint@SQLITE3_3.7.2" = sqlite3BtreeCheckpoint;
+"sqlite3BtreeCursorZero@SQLITE3_3.6.21" = sqlite3BtreeCursorZero;
+"sqlite3BtreeLastPage@SQLITE3_3.7.0.1" = sqlite3BtreeLastPage;
+"sqlite3BtreeSecureDelete@SQLITE3_3.6.23.1" = sqlite3BtreeSecureDelete;
+"sqlite3BtreeSetVersion@SQLITE3_3.7.0.1" = sqlite3BtreeSetVersion;
+"sqlite3CantopenError@SQLITE3_3.6.23.1" = sqlite3CantopenError;
+"sqlite3Checkpoint@SQLITE3_3.7.0.1" = sqlite3Checkpoint;
+"sqlite3CodeRowTriggerDirect@SQLITE3_3.6.19" = sqlite3CodeRowTriggerDirect;
+"sqlite3ConnectionBlocked@SQLITE3_3.6.23.1.1" = sqlite3ConnectionBlocked;
+"sqlite3ConnectionClosed@SQLITE3_3.6.23.1.1" = sqlite3ConnectionClosed;
+"sqlite3ConnectionUnlocked@SQLITE3_3.6.23.1.1" = sqlite3ConnectionUnlocked;
+"sqlite3CorruptError@SQLITE3_3.6.23.1" = sqlite3CorruptError;
+"sqlite3CreateColumnExpr@SQLITE3_3.6.20" = sqlite3CreateColumnExpr;
+"sqlite3ExprCanBeNull@SQLITE3_3.6.21" = sqlite3ExprCanBeNull;
+"sqlite3ExprCodeGetColumnOfTable@SQLITE3_3.7.0.1" = sqlite3ExprCodeGetColumnOfTable;
+"sqlite3ExprCodeIsNullJump@SQLITE3_3.6.21" = sqlite3ExprCodeIsNullJump;
+"sqlite3ExprListCompare@SQLITE3_3.7.0.1" = sqlite3ExprListCompare;
+"sqlite3ExprNeedsNoAffinityChange@SQLITE3_3.6.21" = sqlite3ExprNeedsNoAffinityChange;
+"sqlite3ExprSetCollByToken@SQLITE3_3.7.2" = sqlite3ExprSetCollByToken;
+"sqlite3FkActions@SQLITE3_3.6.19" = sqlite3FkActions;
+"sqlite3FkCheck@SQLITE3_3.6.19" = sqlite3FkCheck;
+"sqlite3FkDelete@SQLITE3_3.6.19" = sqlite3FkDelete;
+"sqlite3FkDropTable@SQLITE3_3.6.19" = sqlite3FkDropTable;
+"sqlite3FkOldmask@SQLITE3_3.6.19" = sqlite3FkOldmask;
+"sqlite3FkReferences@SQLITE3_3.6.19" = sqlite3FkReferences;
+"sqlite3FkRequired@SQLITE3_3.6.19" = sqlite3FkRequired;
+"sqlite3Fts3AllSegdirs@SQLITE3_3.6.21" = sqlite3Fts3AllSegdirs;
+"sqlite3Fts3Dequote@SQLITE3_3.6.21" = sqlite3Fts3Dequote;
+"sqlite3Fts3ExprLoadDoclist@SQLITE3_3.6.22" = sqlite3Fts3ExprLoadDoclist;
+"sqlite3Fts3ExprNearTrim@SQLITE3_3.6.23.1" = sqlite3Fts3ExprNearTrim;
+"sqlite3Fts3FindPositions@SQLITE3_3.6.22" = sqlite3Fts3FindPositions;
+"sqlite3Fts3GetVarint32@SQLITE3_3.6.21" = sqlite3Fts3GetVarint32;
+"sqlite3Fts3GetVarint@SQLITE3_3.6.21" = sqlite3Fts3GetVarint;
+"sqlite3Fts3HashFindElem@SQLITE3_3.6.22" = sqlite3Fts3HashFindElem;
+"sqlite3Fts3InitTokenizer@SQLITE3_3.6.21" = sqlite3Fts3InitTokenizer;
+"sqlite3Fts3Matchinfo@SQLITE3_3.6.22" = sqlite3Fts3Matchinfo;
+"sqlite3Fts3NextToken@SQLITE3_3.6.21" = sqlite3Fts3NextToken;
+"sqlite3Fts3Offsets@SQLITE3_3.6.21" = sqlite3Fts3Offsets;
+"sqlite3Fts3Optimize@SQLITE3_3.6.21" = sqlite3Fts3Optimize;
+"sqlite3Fts3PendingTermsClear@SQLITE3_3.6.21" = sqlite3Fts3PendingTermsClear;
+"sqlite3Fts3PendingTermsFlush@SQLITE3_3.6.21" = sqlite3Fts3PendingTermsFlush;
+"sqlite3Fts3PutVarint@SQLITE3_3.6.21" = sqlite3Fts3PutVarint;
+"sqlite3Fts3ReadBlock@SQLITE3_3.6.21" = sqlite3Fts3ReadBlock;
+"sqlite3Fts3SegReaderFree@SQLITE3_3.6.21" = sqlite3Fts3SegReaderFree;
+"sqlite3Fts3SegReaderIterate@SQLITE3_3.6.21" = sqlite3Fts3SegReaderIterate;
+"sqlite3Fts3SegReaderNew@SQLITE3_3.6.21" = sqlite3Fts3SegReaderNew;
+"sqlite3Fts3SegReaderPending@SQLITE3_3.6.22" = sqlite3Fts3SegReaderPending;
+"sqlite3Fts3Snippet@SQLITE3_3.6.21" = sqlite3Fts3Snippet;
+"sqlite3Fts3UpdateMethod@SQLITE3_3.6.21" = sqlite3Fts3UpdateMethod;
+"sqlite3Fts3VarintLen@SQLITE3_3.6.21" = sqlite3Fts3VarintLen;
+"sqlite3IntTokens@SQLITE3_3.7.0.1" = sqlite3IntTokens;
+"sqlite3JournalModename@SQLITE3_3.7.0.1" = sqlite3JournalModename;
+"sqlite3MisuseError@SQLITE3_3.6.23.1" = sqlite3MisuseError;
+"sqlite3MultiWrite@SQLITE3_3.6.19" = sqlite3MultiWrite;
+"sqlite3NoopMutex@SQLITE3_3.7.0.1" = sqlite3NoopMutex;
+"sqlite3OsCurrentTimeInt64@SQLITE3_3.7.0.1" = sqlite3OsCurrentTimeInt64;
+"sqlite3OsShmBarrier@SQLITE3_3.7.0.1" = sqlite3OsShmBarrier;
+"sqlite3OsShmLock@SQLITE3_3.7.0.1" = sqlite3OsShmLock;
+"sqlite3OsShmMap@SQLITE3_3.7.0.1" = sqlite3OsShmMap;
+"sqlite3OsShmUnmap@SQLITE3_3.7.0.1" = sqlite3OsShmUnmap;
+"sqlite3PagerCheckpoint@SQLITE3_3.7.0.1" = sqlite3PagerCheckpoint;
+"sqlite3PagerCloseWal@SQLITE3_3.7.0.1" = sqlite3PagerCloseWal;
+"sqlite3PagerDontWrite@SQLITE3_3.6.23.1" = sqlite3PagerDontWrite;
+"sqlite3PagerExclusiveLock@SQLITE3_3.7.2" = sqlite3PagerExclusiveLock;
+"sqlite3PagerGetJournalMode@SQLITE3_3.7.0.1" = sqlite3PagerGetJournalMode;
+"sqlite3PagerMemUsed@SQLITE3_3.7.0.1" = sqlite3PagerMemUsed;
+"sqlite3PagerOkToChangeJournalMode@SQLITE3_3.7.0.1" = sqlite3PagerOkToChangeJournalMode;
+"sqlite3PagerOpenWal@SQLITE3_3.7.0.1" = sqlite3PagerOpenWal;
+"sqlite3PagerSetJournalMode@SQLITE3_3.7.0.1" = sqlite3PagerSetJournalMode;
+"sqlite3PagerWalCallback@SQLITE3_3.7.0.1" = sqlite3PagerWalCallback;
+"sqlite3PagerWalSupported@SQLITE3_3.7.0.1" = sqlite3PagerWalSupported;
+"sqlite3TriggerColmask@SQLITE3_3.6.21" = sqlite3TriggerColmask;
+"sqlite3VdbeAddOp4Int@SQLITE3_3.6.21" = sqlite3VdbeAddOp4Int;
+"sqlite3VdbeCheckFk@SQLITE3_3.6.19" = sqlite3VdbeCheckFk;
+"sqlite3VdbeDeleteObject@SQLITE3_3.7.2" = sqlite3VdbeDeleteObject;
+"sqlite3VdbeExpandSql@SQLITE3_3.6.21" = sqlite3VdbeExpandSql;
+"sqlite3VdbeGetValue@SQLITE3_3.6.20" = sqlite3VdbeGetValue;
+"sqlite3VdbeLinkSubProgram@SQLITE3_3.7.2" = sqlite3VdbeLinkSubProgram;
+"sqlite3VdbeMemStoreType@SQLITE3_3.6.20" = sqlite3VdbeMemStoreType;
+"sqlite3VdbeParameterIndex@SQLITE3_3.6.21" = sqlite3VdbeParameterIndex;
+"sqlite3VdbeRunOnlyOnce@SQLITE3_3.6.23.1" = sqlite3VdbeRunOnlyOnce;
+"sqlite3VdbeSetVarmask@SQLITE3_3.6.20" = sqlite3VdbeSetVarmask;
+"sqlite3WalBeginReadTransaction@SQLITE3_3.7.0.1" = sqlite3WalBeginReadTransaction;
+"sqlite3WalBeginWriteTransaction@SQLITE3_3.7.0.1" = sqlite3WalBeginWriteTransaction;
+"sqlite3WalCallback@SQLITE3_3.7.0.1" = sqlite3WalCallback;
+"sqlite3WalCheckpoint@SQLITE3_3.7.0.1" = sqlite3WalCheckpoint;
+"sqlite3WalClose@SQLITE3_3.7.0.1" = sqlite3WalClose;
+"sqlite3WalDbsize@SQLITE3_3.7.0.1" = sqlite3WalDbsize;
+"sqlite3WalDefaultHook@SQLITE3_3.7.0.1" = sqlite3WalDefaultHook;
+"sqlite3WalEndReadTransaction@SQLITE3_3.7.0.1" = sqlite3WalEndReadTransaction;
+"sqlite3WalEndWriteTransaction@SQLITE3_3.7.0.1" = sqlite3WalEndWriteTransaction;
+"sqlite3WalExclusiveMode@SQLITE3_3.7.0.1" = sqlite3WalExclusiveMode;
+"sqlite3WalFrames@SQLITE3_3.7.0.1" = sqlite3WalFrames;
+"sqlite3WalOpen@SQLITE3_3.7.0.1" = sqlite3WalOpen;
+"sqlite3WalRead@SQLITE3_3.7.0.1" = sqlite3WalRead;
+"sqlite3WalSavepoint@SQLITE3_3.7.0.1" = sqlite3WalSavepoint;
+"sqlite3WalSavepointUndo@SQLITE3_3.7.0.1" = sqlite3WalSavepointUndo;
+"sqlite3WalUndo@SQLITE3_3.7.0.1" = sqlite3WalUndo;
+"sqlite3XPrintf@SQLITE3_3.6.21" = sqlite3XPrintf;
+"sqlite3_auto_extension@SQLITE_3.3.8" = sqlite3_auto_extension;
+"sqlite3_backup_finish@SQLITE3_3.6.16" = sqlite3_backup_finish;
+"sqlite3_backup_init@SQLITE3_3.6.16" = sqlite3_backup_init;
+"sqlite3_backup_pagecount@SQLITE3_3.6.16" = sqlite3_backup_pagecount;
+"sqlite3_backup_remaining@SQLITE3_3.6.16" = sqlite3_backup_remaining;
+"sqlite3_backup_step@SQLITE3_3.6.16" = sqlite3_backup_step;
+"sqlite3_bind_value@SQLITE_3.3.7" = sqlite3_bind_value;
+"sqlite3_bind_zeroblob@SQLITE_3.4.0" = sqlite3_bind_zeroblob;
+"sqlite3_blob_bytes@SQLITE_3.4.0" = sqlite3_blob_bytes;
+"sqlite3_blob_close@SQLITE_3.4.0" = sqlite3_blob_close;
+"sqlite3_blob_open@SQLITE_3.4.0" = sqlite3_blob_open;
+"sqlite3_blob_read@SQLITE_3.4.0" = sqlite3_blob_read;
+"sqlite3_blob_write@SQLITE_3.4.0" = sqlite3_blob_write;
+"sqlite3_clear_bindings@SQLITE_3.3.7" = sqlite3_clear_bindings;
+"sqlite3_column_database_name16@SQLITE3_COLUMN_METADATA" = sqlite3_column_database_name16;
+"sqlite3_column_database_name@SQLITE3_COLUMN_METADATA" = sqlite3_column_database_name;
+"sqlite3_column_origin_name16@SQLITE3_COLUMN_METADATA" = sqlite3_column_origin_name16;
+"sqlite3_column_origin_name@SQLITE3_COLUMN_METADATA" = sqlite3_column_origin_name;
+"sqlite3_column_table_name16@SQLITE3_COLUMN_METADATA" = sqlite3_column_table_name16;
+"sqlite3_column_table_name@SQLITE3_COLUMN_METADATA" = sqlite3_column_table_name;
+"sqlite3_column_value@SQLITE_3.3.7" = sqlite3_column_value;
+"sqlite3_compileoption_get@SQLITE3_3.6.23.1" = sqlite3_compileoption_get;
+"sqlite3_compileoption_used@SQLITE3_3.6.23.1" = sqlite3_compileoption_used;
+"sqlite3_config@SQLITE3_3.6.16" = sqlite3_config;
+"sqlite3_context_db_handle@SQLITE_3.5.8" = sqlite3_context_db_handle;
+"sqlite3_create_collation_v2@SQLITE_3.4.0" = sqlite3_create_collation_v2;
+"sqlite3_create_module@SQLITE_3.3.7" = sqlite3_create_module;
+"sqlite3_create_module_v2@SQLITE_3.4.1" = sqlite3_create_module_v2;
+"sqlite3_db_config@SQLITE3_3.6.16" = sqlite3_db_config;
+"sqlite3_db_handle@SQLITE_3.2.2" = sqlite3_db_handle;
+"sqlite3_db_mutex@SQLITE3_3.6.16" = sqlite3_db_mutex;
+"sqlite3_db_status@SQLITE3_3.6.16" = sqlite3_db_status;
+"sqlite3_declare_vtab@SQLITE_3.3.7" = sqlite3_declare_vtab;
+"sqlite3_enable_load_extension@SQLITE_3.3.7" = sqlite3_enable_load_extension;
+"sqlite3_enable_shared_cache@SQLITE_3.3.0" = sqlite3_enable_shared_cache;
+"sqlite3_extended_errcode@SQLITE3_3.6.16" = sqlite3_extended_errcode;
+"sqlite3_extended_result_codes@SQLITE_3.3.8" = sqlite3_extended_result_codes;
+"sqlite3_file_control@SQLITE_3.5.0" = sqlite3_file_control;
+"sqlite3_get_autocommit@SQLITE_3.2.2" = sqlite3_get_autocommit;
+"sqlite3_initialize@SQLITE3_3.6.16" = sqlite3_initialize;
+"sqlite3_limit@SQLITE_3.5.8" = sqlite3_limit;
+"sqlite3_load_extension@SQLITE_3.3.7" = sqlite3_load_extension;
+"sqlite3_log@SQLITE3_3.6.23.1" = sqlite3_log;
+"sqlite3_malloc@SQLITE_3.3.7" = sqlite3_malloc;
+"sqlite3_memory_alarm@SQLITE_3.5.0" = sqlite3_memory_alarm;
+"sqlite3_memory_highwater@SQLITE_3.5.0" = sqlite3_memory_highwater;
+"sqlite3_memory_used@SQLITE_3.5.0" = sqlite3_memory_used;
+"sqlite3_mutex_alloc@SQLITE_3.5.0" = sqlite3_mutex_alloc;
+"sqlite3_mutex_enter@SQLITE_3.5.0" = sqlite3_mutex_enter;
+"sqlite3_mutex_free@SQLITE_3.5.0" = sqlite3_mutex_free;
+"sqlite3_mutex_leave@SQLITE_3.5.0" = sqlite3_mutex_leave;
+"sqlite3_mutex_try@SQLITE_3.5.0" = sqlite3_mutex_try;
+"sqlite3_next_stmt@SQLITE3_3.6.16" = sqlite3_next_stmt;
+"sqlite3_open_v2@SQLITE_3.5.0" = sqlite3_open_v2;
+"sqlite3_os_end@SQLITE3_3.6.16" = sqlite3_os_end;
+"sqlite3_os_init@SQLITE3_3.6.16" = sqlite3_os_init;
+"sqlite3_overload_function@SQLITE_3.3.8" = sqlite3_overload_function;
+"sqlite3_prepare16_v2@SQLITE_3.3.9" = sqlite3_prepare16_v2;
+"sqlite3_prepare_v2@SQLITE_3.3.9" = sqlite3_prepare_v2;
+"sqlite3_profile@SQLITE_3.2.6" = sqlite3_profile;
+"sqlite3_randomness@SQLITE_3.5.8" = sqlite3_randomness;
+"sqlite3_realloc@SQLITE_3.3.7" = sqlite3_realloc;
+"sqlite3_release_memory@SQLITE_3.5.0" = sqlite3_release_memory;
+"sqlite3_reset_auto_extension@SQLITE_3.3.8" = sqlite3_reset_auto_extension;
+"sqlite3_result_error_code@SQLITE_3.5.6" = sqlite3_result_error_code;
+"sqlite3_result_error_nomem@SQLITE_3.5.0" = sqlite3_result_error_nomem;
+"sqlite3_result_error_toobig@SQLITE_3.4.0" = sqlite3_result_error_toobig;
+"sqlite3_result_zeroblob@SQLITE_3.4.0" = sqlite3_result_zeroblob;
+"sqlite3_rollback_hook@SQLITE_3.3.0" = sqlite3_rollback_hook;
+"sqlite3_shutdown@SQLITE3_3.6.16" = sqlite3_shutdown;
+"sqlite3_sleep@SQLITE_3.3.7" = sqlite3_sleep;
+"sqlite3_soft_heap_limit@SQLITE_3.5.0" = sqlite3_soft_heap_limit;
+"sqlite3_sourceid@SQLITE3_3.6.18" = sqlite3_sourceid;
+"sqlite3_sql@SQLITE_3.5.3" = sqlite3_sql;
+"sqlite3_status@SQLITE3_3.6.16" = sqlite3_status;
+"sqlite3_stmt_status@SQLITE3_3.6.16" = sqlite3_stmt_status;
+"sqlite3_strnicmp@SQLITE3_3.6.17" = sqlite3_strnicmp;
+"sqlite3_table_column_metadata@SQLITE3_COLUMN_METADATA" = sqlite3_table_column_metadata;
+"sqlite3_test_control@SQLITE_3.5.5" = sqlite3_test_control;
+"sqlite3_thread_cleanup@SQLITE_3.3.1" = sqlite3_thread_cleanup;
+"sqlite3_threadsafe@SQLITE_3.5.0" = sqlite3_threadsafe;
+"sqlite3_transfer_bindings@SQLITE_3.2.2" = sqlite3_transfer_bindings;
+"sqlite3_unlock_notify@SQLITE3_3.6.23.1.1" = sqlite3_unlock_notify;
+"sqlite3_update_hook@SQLITE_3.3.0" = sqlite3_update_hook;
+"sqlite3_value_numeric_type@SQLITE_3.3.4" = sqlite3_value_numeric_type;
+"sqlite3_vfs_find@SQLITE_3.5.0" = sqlite3_vfs_find;
+"sqlite3_vfs_register@SQLITE_3.5.0" = sqlite3_vfs_register;
+"sqlite3_vfs_unregister@SQLITE_3.5.0" = sqlite3_vfs_unregister;
+"sqlite3_wal_autocheckpoint@SQLITE3_3.7.0.1" = sqlite3_wal_autocheckpoint;
+"sqlite3_wal_checkpoint@SQLITE3_3.7.0.1" = sqlite3_wal_checkpoint;
+"sqlite3_wal_hook@SQLITE3_3.7.0.1" = sqlite3_wal_hook;
--- sqlite-3.7.4/libsqlite3.map-	2011-02-15 20:04:12.665295871 +0300
+++ sqlite-3.7.4/libsqlite3.map	2011-02-15 21:33:32.328441621 +0300
@@ -0,0 +1,56 @@
+SQLITE_3.2.2 {
+    local:
+	[^s]*;
+};
+SQLITE_3.2.6 {
+};
+SQLITE_3.3.0 {
+};
+SQLITE_3.3.1 {
+};
+SQLITE_3.3.4 {
+};
+SQLITE_3.3.7 {
+};
+SQLITE_3.3.8 {
+};
+SQLITE_3.3.9 {
+};
+SQLITE_3.4.0 {
+};
+SQLITE_3.4.1 {
+};
+SQLITE_3.5.0 {
+};
+SQLITE_3.5.3 {
+};
+SQLITE_3.5.5 {
+};
+SQLITE_3.5.6 {
+};
+SQLITE_3.5.8 {
+};
+SQLITE3_3.6.16 {
+};
+SQLITE3_3.6.17 {
+};
+SQLITE3_3.6.18 {
+};
+SQLITE3_3.6.19 {
+};
+SQLITE3_3.6.20 {
+};
+SQLITE3_3.6.21 {
+};
+SQLITE3_3.6.22 {
+};
+SQLITE3_3.6.23.1 {
+};
+SQLITE3_3.6.23.1.1 {
+};
+SQLITE3_3.7.0.1 {
+};
+SQLITE3_3.7.2 {
+};
+SQLITE3_COLUMN_METADATA {
+};
