--- sqlite-3.6.19/docsrc/main.mk.alt	2009-10-19 14:48:03 +0000
+++ sqlite-3.6.19/docsrc/main.mk	2009-10-19 15:35:56 +0000
@@ -21,7 +21,7 @@
 # build the SQLite library and testing tools.
 ################################################################################
 
-all:	base evidence matrix doc
+all:	base doc
 
 sqlite3.h:	$(SRC)/src/sqlite.h.in $(SRC)/manifest.uuid $(SRC)/VERSION
 	$(TCLSH) $(SRC)/tool/mksqlite3h.tcl $(SRC) | \
